tests/src/test/java/alluxio/testutils/LocalAlluxioClusterResource.java

```diff
@@ -268,6 +274,7 @@ public Statement apply(Statement statement, Description description) {
       return new Statement() {
         @Override
         public void evaluate() throws Throwable {
+          AlluxioProperties props = ServerConfiguration.copyProperties();
```

Comment:
instead of copying the whole thing, I tried using an annotation, similar to the existing one. This approach might be cleaner and more similar to what existed?
```java
  /**
   * An annotation for test methods that can be used to override any user configuration values. This
   * should not be used with server properties, since they may not take effect if the cluster is
   * a {@code @ClassRule} (single cluster per test file).
   */
  @Retention(RetentionPolicy.RUNTIME)
  public @interface UserConfig {
    String[] confParams() default {};
  }
```

Then, here, we examine the annotation and save the old values and set the new values:
```java
            Annotation configAnnotation = description.getAnnotation(UserConfig.class);
            if (configAnnotation != null) {
              UserConfig config = (UserConfig) configAnnotation;
              mOldConf.clear();
              mOldUnset.clear();
              // Override the configuration parameters with any configuration params
              for (int i = 0; i < config.confParams().length; i += 2) {
                PropertyKey prop = PropertyKey.fromString(config.confParams()[i]);
                if (ServerConfiguration.isSet(prop)) {
                  // save the existing conf value
                  mOldConf.put(prop, ServerConfiguration.get(prop));
                } else {
                  mOldUnset.add(prop);
                }
                // set the new conf value
                ServerConfiguration.set(prop, config.confParams()[i + 1]);
              }
            }
```

and in the `finally`, we set then back:
```java
            if (!mOldConf.isEmpty()) {
              // reset the old conf values
              for (Map.Entry<PropertyKey, String> entry : mOldConf.entrySet()) {
                ServerConfiguration.set(entry.getKey(), entry.getValue());
              }
              mOldConf.clear();
            }
            if (!mOldUnset.isEmpty()) {
              // reset the old unset conf values
              for (PropertyKey prop : mOldUnset) {
                ServerConfiguration.unset(prop);
              }
              mOldUnset.clear();
            }
```